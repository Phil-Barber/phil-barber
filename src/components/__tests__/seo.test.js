import React from 'react';
import { shallow } from 'enzyme';
import SEO from '../seo';
import * as Gatsby from 'gatsby';

describe('SEO', () => {
  const siteMetadata = {
    title: 'Test title template',
    description: 'Test description',
    author: 'Test author',
  };
  Gatsby.useStaticQuery = jest.fn(() => ({ site: { siteMetadata } }));
  const defaultProps = {
    title: 'Test title',
  };

  describe('defaultProps', () => {
    it('Matches snapshot', () => {
      const component = shallow(<SEO {...defaultProps} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('metaDescription', () => {
    const getDescription = (component) =>
      component.props().meta.find(({ name }) => name === 'description').content;

    it('uses metaDataDescription if description prop not provided', () => {
      const component = shallow(<SEO {...defaultProps} />);
      expect(getDescription(component)).toBe(siteMetadata.description);
    });

    it('uses description prop if provided', () => {
      const description = 'Real description';
      const component = shallow(
        <SEO {...defaultProps} description={description} />
      );
      expect(getDescription(component)).toBe(description);
    });
  });

  it('includes additional meta fields', () => {
    const additionalField = { name: 'addtional', content: 'field' };
    const component = shallow(
      <SEO {...defaultProps} meta={[additionalField]} />
    );
    const metaProp = component.props().meta;
    expect(metaProp.length).toBeGreaterThan(1);
    expect(metaProp).toEqual(expect.arrayContaining([additionalField]));
  });
});
