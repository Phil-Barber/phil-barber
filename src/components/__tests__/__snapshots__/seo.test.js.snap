// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO defaultProps Matches snapshot 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en",
    }
  }
  meta={
    Array [
      Object {
        "content": "Test description",
        "name": "description",
      },
      Object {
        "content": "Test title",
        "property": "og:title",
      },
      Object {
        "content": "Test description",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "Test author",
        "name": "twitter:creator",
      },
      Object {
        "content": "Test title",
        "name": "twitter:title",
      },
      Object {
        "content": "Test description",
        "name": "twitter:description",
      },
    ]
  }
  title="Test title"
  titleTemplate="%s | Test title template"
/>
`;
